### PR TITLE
Adding an function call to keep the ptr alive from SROA

### PR DESCRIPTION
**Pull request by zhengyang92**

### CODE REVIEW

**Comment by zhengyang92** [comment]
This patch also
 deletes the unnecessary trailing spaces in souper2llvm.in

File: utils/souper2llvm.in [hunk 1]
```diff
@@ -611,10 +618,13 @@ def printFuncHeader():
             ptr = "%ptr_" + str(MDRangePtr)
             masks = "  " + ptr + " = alloca " + w + "\n" +\
             "  store " + w + " " + name + ", " + w + "* " + ptr + "\n" +\
+            "  call void @fun_" + str(functionDeclCount) + "(" + w + "* " + ptr + ")\n" +\
             "  " + old + " = load " + w + ", " + w + "* " + ptr + ", !range !" + str(MDRangePtr) + "\n" +\
             "  store " + w + " " + old + ", " + w + "* @glob_" + regnum + "_nonZero\n" + masks
             metadataStr = metadataStr + "!" + str(MDRangePtr) + " = " + "!{ " + w + " 1" + ", " + w + " 0"+ " }\n"
             MDRangePtr += 1
+            functionDeclStr += "declare void @fun_" + str(functionDeclCount) + "(" + w + "* " + ptr + ")\n";
```
Comment (jubitaneja) [inline utils/souper2llvm.in:626]: I'd suggest to take off the semi-colon at the end. Though it won't make a difference here in this case. Just a nitpick.

File: utils/souper2llvm.in [hunk 2]
```diff
@@ -596,10 +600,13 @@ def printFuncHeader():
             ptr = "%ptr_" + str(MDRangePtr)
             masks = "  " + ptr + " = alloca " + w + "\n" +\
             "  store " + w + " " + name + ", " + w + "* " + ptr + "\n" +\
+            "  call void @fun_" + str(functionDeclCount) + "(" + w + "* " + ptr + ")\n" +\
             "  " + old + " = load " + w + ", " + w + "* " + ptr + ", !range !" + str(MDRangePtr) + "\n" +\
             "  store " + w + " " + old + ", " + w + "* @glob_" + regnum + "_range\n" + masks
             metadataStr = metadataStr + "!" + str(MDRangePtr) + " = " + "!{ " + w + " " + str(lower[reg]) + ", " + w + " " + str(upper[reg]) + " }\n"
             MDRangePtr += 1
+            functionDeclStr += "declare void @fun_" + str(functionDeclCount) + "(" + w + "* " + ptr + ")\n";
```
Comment (jubitaneja) [inline utils/souper2llvm.in:608]: Likewise, ';' here too.

**Review by jubitaneja — COMMENTED** [review]
That was a smart trick you added. :) Thanks! 
I am merging your patch and will take off that semi-colon by myself. No problem.

**Pull request merged**